typescript/lib/parsers/hf.ts

```diff
@@ -170,7 +240,8 @@ export class HuggingFaceTextGenerationParser extends ParameterizedModelParser<Te
     }
 
     if (output.output_type === "execute_result") {
-      return output.data as string;
+      return (output.data as TextGenerationOutput | TextGenerationStreamOutput)
+        .generated_text as string;
```

Comment:
I also noticed this bug, however I think this should be fixed when constructing the result, (from my wip diff)[https://github.com/lastmile-ai/aiconfig/pull/213/files#diff-a31886f1d5938735e6e6d480f8a8dadb925d6ae6163f953db5cf7c56cb82201cR95-R109]

- Stream output to return accumulated message, (not last token)[https://github.com/lastmile-ai/aiconfig/pull/213/files#diff-a31886f1d5938735e6e6d480f8a8dadb925d6ae6163f953db5cf7c56cb82201cR95]
- regular output to set data to the string .generated_text (link)[https://github.com/lastmile-ai/aiconfig/pull/213/files#diff-a31886f1d5938735e6e6d480f8a8dadb925d6ae6163f953db5cf7c56cb82201cR115]

Reply:
I think it's correct for the output data to match the response structure, instead of pulling the raw text out of it, fwiw. In any case, this is just fixing the existing code to not type the object as string so going to land as-is and any updates to the result in your PR can be propagated back to the core class as well